presto-main/src/main/java/io/prestosql/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -166,19 +170,26 @@ private void validateShowStatsSubquery(ShowStats node, Query query, QuerySpecifi
 
         private Node rewriteShowStats(ShowStats node, Table table, List<SelectItem> selectItems, Constraint constraint)
         {
-            List<String> statsColumnNames = buildColumnsNames();
             TableHandle tableHandle = getTableHandle(node, table.getName());
             TableStatistics tableStatistics = metadata.getTableStatistics(session, tableHandle, constraint);
             TableMetadata tableMetadata = metadata.getTableMetadata(session, tableHandle);
 
             Map<String, ColumnHandle> columnHandles = metadata.getColumnHandles(session, tableHandle);
             Set<String> extractNamedResultColumns = extractStatsColumns(tableMetadata, selectItems);
+
+            try {
+                accessControl.checkCanSelectFromColumns(session.toSecurityContext(), tableMetadata.getQualifiedName(), extractNamedResultColumns);
+            }
+            catch (AccessDeniedException e) {
+                throw new AccessDeniedException(format("Cannot show stats for table %s", tableMetadata.getTable().toString()));
+            }
+
             List<Expression> resultRows = buildStatisticsRows(tableMetadata, columnHandles, extractNamedResultColumns, tableStatistics);
 
-            return simpleQuery(selectAll(buildSelectItems(statsColumnNames)),
+            return simpleQuery(selectAll(buildSelectItems()),
```

Comment:
unrelated to introduction of access control;

btw why aren't you passing `statsColumnNames` anymore?

Reply:
```
 return buildColumnsNames().stream()
```